### DISCUSSION TITLE
9.4 Env Issues

### DISCUSSION THREAD

**Discussion by waltercolindres** [vercel/vercel#4420] [Help] [answered]
I've been trying to piece this together but I am stuck on the env setup.

I can get the site to build on vercel but the API call is failing. Works on local though. 

vercel.json
`{
  "build": {
    "env": {
      "CONTENTFUL_SPACE_ID": "@contentful_space_id",
      "CONTENTFUL_ACCESS_TOKEN": "@contentful_access_token"
    }
  },
  "github": {
    "enabled": true
  }
}
`

contentful service
`const client = require("contentful").createClient({
  space: process.env.NEXT_PUBLIC_CONTENTFUL_SPACE_ID,
  accessToken: process.env.NEXT_PUBLIC_CONTENTFUL_ACCESS_TOKEN,
  resolveLinks: true,
});`

env.production
`
NEXT_PUBLIC_CONTENTFUL_SPACE_ID=@contentful_space_id
NEXT_PUBLIC_CONTENTFUL_ACCESS_TOKEN=@contentful_access_token
`
I am bit unclear on when I would want to use NEXT_PUBLIC_ . The site will fail locally without it (missing accessToken).

When I make a request to contentful from prod the call includes @contentful_space_id vs the variable value.

path: /spaces/@contentful_space_id/environments/master/entries?include=3&sys.id=XXXXX
authorization: Bearer @contentful_access_token

Thanks in advance!

**Comment by paulogdm** [comment, marked as answer]
Please use our newly release [environments UI](https://vercel.com/blog/environment-variables-ui) instead of `vercel.json`. Any variables with `NEXT_PUBLIC_` will be added automatically to your frontend bundle, while other variables can just be used at build time.

Can you try it and let me know?

**Reply by waltercolindres** [reply]
I feel like I tried this a hundred different ways! It works now. Just needed your magic! Thank you.
